### ISSUE TITLE
CKEditor does not allow added content after a left or right aligned image.

### ISSUE THREAD

**Issue by Graham-72** [backdrop/backdrop-issues#3282]
## Describe your issue
When adding content in a WYSIWYG field one may choose to add an image and set it to be left or right aligned. However, if you do this, CKEditor does not allow the insertion point to move below the image and further content cannot be entered.

### Steps to reproduce (if reporting a bug)
Create some content and add an image to the text. When adding the image select left or right alignment. Then try to add text or another image below this one. Even after saving this is not possible.

### Relevant version/system information (if applicable)
This is happening with release 1.10.1

**Comment by herbdool** [comment]
I'm not able to replicate it with 1.11.x-dev if I understand your steps correctly.

**Comment by Graham-72** [comment]
I still have this problem with 1.11.0 and I have tried with both Firefox and Chrome.

My steps are:
1. Create a new post.
2. Using WYSIWYG add a line or two of text.
3. Add an image, uploaded from my PC and select a width and also align left, so that the img source includes data-align="left".
4. Save post.
5. Re-open post to edit it.

At this point it is impossible in WYSIWYG to place the insertion point below the image in order to add more to the post below the image.

**Comment by herbdool** [comment]
I'm not sure I understand but seems like that's what is supposed to happen. It will put the text to the right of the image and wrap below it if there is enough text.

**Comment by herbdool** [comment]
The text wrapping around the image is a usability issue?

Maybe you post screenshots or a gif because I'm still not clear on the issue.

**Comment by olafgrabienski** [comment]
The behaviour in CKEditor is indeed a bit strange. I've followed the five steps described in https://github.com/backdrop/backdrop-issues/issues/3282#issuecomment-421896701. At this point, I wasn't able, using the mouse, to put the cursor *below* the image. Ok, I guess that's intended because of the left alignment (floating which isn't only represented on the page but also in the CKEditor area). However, it was also not possible to place the cursor *beside* the image, at least when I used the mouse. Consequently, I wasn't able to add more content 'after' the image, and I think that's the issue here.

As a workaround, I was able to move the cursor from the introductory text to the image and further using the right arrow of my keyboard.

**Comment by Graham-72** [comment]
Yes, I described the problem badly. As Olaf says, the problem is that in certain circumstances it becomes impossible to find/place the insertion point and hence to add text. I think perhaps this arises if the user adds an image before adding the associated text. The result will be an image within paragraph tags, but the impossibility of placing the insertion point within this paragraph without resorting to source code.

**Comment by Graham-72** [comment]
I am now experiencing a different version of this problem, regardless of the aligment (or none) of the image. The user experience seems different in Firefox and Chrome.

If I enter a lone image in Firefox, with no preceding text, there is no sign of an insertion point (caret) and I cannot get it to appear. I suspect it may be outside the viewport.

If I do the same thing in Chrome, the insertion point is not immediately visible but appears if I click in the empty space to the right of the image. However, it appears at the extreme right of the viewport, level with the top of the image. If I then create a new line by using my enter key, the insertion point moves below the image, a new empty paragraph having been created.

I would prefer it if the caret were to appear immediately to the right of the image and without having to click in the space to make it visible.

**Comment by Graham-72** [comment]
@klonos I have just tried this with CKEditor 4.10.1 with your PR #2361 sandbox and I have the same problem in Firefox (version 63.0.3 in Windows 10) with no visible insertion point. However, using Chrome the insertion point now appears where I expect it to, immediately at the top right of the image. Does it do the same for you? If so, should this be a new issue?
